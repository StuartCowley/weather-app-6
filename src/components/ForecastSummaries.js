import PropTypes from "prop-types";

import ForecastSummary from "./ForecastSummary";

function ForecastSummaries({ forecasts }) {
  return (
    <section className="ForecastSummaries">
      {forecasts.map((forecast) => (
        <ForecastSummary
          key={forecast.date}
          date={forecast.date}
          icon={forecast.icon}
          temperature={forecast.temperature}
          description={forecast.description}
        />
      ))}
    </section>
  );
}

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number.isRequired,
      icon: PropTypes.string.isRequired,
      temperature: PropTypes.shape({
        min: PropTypes.number,
        max: PropTypes.number.isRequired,
      }).isRequired,
      description: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default ForecastSummaries;
