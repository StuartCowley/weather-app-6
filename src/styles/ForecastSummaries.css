.forecast-summaries {
  display: flex;
  flex-flow: row nowrap;
  gap: 2rem;
  margin-top: 2.5rem;
}

.forecast-summaries * {
  flex-grow: 1;
}
