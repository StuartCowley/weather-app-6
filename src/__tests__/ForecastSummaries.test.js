import { render } from "@testing-library/react";

import ForecastSummaries from "../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 1111111,
        icon: 800,
        temperature: {
          min: 10,
          max: 20,
        },
        description: "Test Description",
      },
      {
        date: 1648376637,
        icon: 602,
        temperature: {
          min: 10,
          max: 20,
        },
        description: "Test Description",
      },
    ],
    handleForecastSelect: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        handleForecastSelect={validProps.handleForecastSelect}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        handleForecastSelect={validProps.handleForecastSelect}
      />
    );

    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
