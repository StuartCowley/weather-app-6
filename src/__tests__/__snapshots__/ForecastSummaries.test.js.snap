// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <section
    class="forecast-summaries"
  >
    <article
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        <p
          class="forecast-summary__date__content"
        >
          Thu 1st Jan
        </p>
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        <p
          class="forecast-summary__temperature__content"
        >
          20 °C
        </p>
      </div>
      <div
        class="forecast-summary__description"
      >
        <p
          class="forecast-summary__description__content"
        >
          Test Description
        </p>
      </div>
      <div
        class="forecast-summary__more-details"
      >
        <button
          class="forecast-summary__more-details__button"
          data-date="1111111"
          type="button"
        >
          More Details
        </button>
      </div>
    </article>
    <article
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        <p
          class="forecast-summary__date__content"
        >
          Tue 20th Jan
        </p>
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-snow"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        <p
          class="forecast-summary__temperature__content"
        >
          20 °C
        </p>
      </div>
      <div
        class="forecast-summary__description"
      >
        <p
          class="forecast-summary__description__content"
        >
          Test Description
        </p>
      </div>
      <div
        class="forecast-summary__more-details"
      >
        <button
          class="forecast-summary__more-details__button"
          data-date="1648376637"
          type="button"
        >
          More Details
        </button>
      </div>
    </article>
  </section>
</DocumentFragment>
`;
